src/controllers/continuousUpdates/continuousUpdates.ts

```diff
@@ -0,0 +1,272 @@
+    )
+    this.accountStateLatestInterval.start()
+
+    this.accountStatePendingInterval = createRecurringTimeout(
+      async () => {
+        if (!this.#main.selectedAccount.account) {
+          console.error('No selected account to update pending state')
+          return
+        }
+
+        const networksToUpdate = this.#main.activity.broadcastedButNotConfirmed
+          .map((op) => op.chainId)
+          .filter((chainId, index, self) => self.indexOf(chainId) === index)
+
+        if (!networksToUpdate.length) {
+          this.accountStatePendingInterval.stop()
+          this.accountStateLatestInterval.restart()
+          return
+        }
+
+        await this.#main.accounts.updateAccountState(
+          this.#main.selectedAccount.account.addr,
+          'pending',
+          networksToUpdate
+        )
+
+        const newestOpTimestamp = this.#main.activity.broadcastedButNotConfirmed.reduce(
+          (newestTimestamp, accOp) => {
+            return accOp.timestamp > newestTimestamp ? accOp.timestamp : newestTimestamp
+          },
+          0
+        )
+        const interval = getAccountOpsIntervalRefreshTime(
+          ACCOUNT_STATE_PENDING_INTERVAL,
+          newestOpTimestamp
+        )
+        this.accountStatePendingInterval.updateTimeout({ timeout: interval })
+      },
+      ACCOUNT_STATE_PENDING_INTERVAL,
+      this.emitError.bind(this)
+    )
+
+    this.accountsOpsStatusesInterval = createRecurringTimeout(
+      async () => {
+        const { newestOpTimestamp } = await this.#main.updateAccountsOpsStatuses()
+        // Schedule the next update only when the previous one completes
+        const interval = getAccountOpsIntervalRefreshTime(
+          ACTIVITY_REFRESH_INTERVAL,
+          newestOpTimestamp
+        )
+        this.accountsOpsStatusesInterval.updateTimeout({ timeout: interval })
+      },
+      ACTIVITY_REFRESH_INTERVAL,
+      this.emitError.bind(this)
+    )
+
+    /**
+     * Update failed network states more often. If a network's first failed
+     *  update is just now, retry in 8s. If it's a repeated failure, retry in 20s.
+     */
+    this.fastAccountStateReFetchTimeout = createRecurringTimeout(
+      async () => {
+        const failedChainIds = getNetworksWithFailedRPC({
+          providers: this.#main.providers.providers
+        })
+        if (!failedChainIds.length) {
+          this.fastAccountStateReFetchTimeout.stop()
+          this.#retriedFastAccountStateReFetchForNetworks = []
+          return
+        }
+
+        await this.#main.accounts.updateAccountStates(
+          this.#main.selectedAccount.account?.addr,
+          'latest',
+          failedChainIds.map((id) => BigInt(id))
+        )
+
+        const failedChainIdsAfterUpdate = getNetworksWithFailedRPC({
+          providers: this.#main.providers.providers
+        })
+
+        const retriedFastAccountStateReFetchForNetworks =
+          this.#retriedFastAccountStateReFetchForNetworks
```

Comment:
We have to push these ids to `retriedFastAccountStateReFetchForNetworks`. Otherwise, if the account state update succeeds on the first try (after 8s), we won't update the portfolio

Reply:
Also `recentlyFailedNetworks` will be incorrect if we don't do this